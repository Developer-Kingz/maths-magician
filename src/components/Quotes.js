import React, { useEffect, useState } from 'react';

const Quotes = () => {
  const [quotes, setQuotes] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes',
          {
            headers: {
              'X-Api-Key':
                                'Brp9Cj/1tTEh3HQN6E8ReA==Pnh6iW9UxEXlUpBk',
            },
          },
        );
        const json = await res.json();
        setQuotes(json);
        setLoading(false);
      } catch (error) {
        setError(true);
      }
    };
    setTimeout(fetchData, 5000);
  }, [quotes]);

  return (
    <div className="my-quotes">
      <h1>Quotes of the day</h1>
      <div className="quotes">
        {loading
          ? 'loading...'
          : quotes.map((quote) => (
            <li key={quote.id}>{quote.quote}</li>
          ))}
      </div>
      {error ? <p>Cannot load data</p> : null}
    </div>
  );
};

export default Quotes;
