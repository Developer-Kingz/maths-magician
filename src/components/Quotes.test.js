// Quotes.test.js
import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import Quotes from './Quotes';

test('Quotes component matches snapshot', () => {
  // Wrap Quotes component in MemoryRouter
  const { asFragment } = render(
    <MemoryRouter>
      <Quotes />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
