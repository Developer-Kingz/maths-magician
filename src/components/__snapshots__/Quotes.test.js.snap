// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="navbar"
    >
      <div
        class="logo"
      >
        <h1>
          Math Magician
        </h1>
      </div>
      <div
        class="nav"
      >
        <ul>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <li>
              Home
            </li>
          </a>
          <a
            class=""
            href="/calculator"
          >
            <li>
              Calculator
            </li>
          </a>
          <a
            class=""
            href="/quotes"
          >
            <li>
              Quotes
            </li>
          </a>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="my-quotes"
  >
    <h1>
      Quotes of the day
    </h1>
    <div
      class="quotes"
    />
  </div>
</DocumentFragment>
`;
